### PR TITLE
Implement Custom Isotopes

### PR DESCRIPTION
**Pull request by mgiammar**
The core functionality for creating and getting information from a custom Isotope has been written, although some further design needs to go into how serialized Isotope dictionaries are parsed from a JSON file, particularly for Method objects.

- [x] Fully comment code
- [x] Include test cases for new functions
- [x] Write some online documentation describing custom isotope use
- [x] Update changelog

### CODE REVIEW

File: src/mrsimulator/spin_system/isotope.py [hunk 1]
```diff
@@ -53,7 +55,113 @@ def get_isotope(cls, v, *, values, **kwargs):
...
+        #     "gyromagnetic_ratio": self.gyromagnetic_ratio,
+        #     "quadrupole_moment": self.quadrupole_moment,
+        #     "atomic_number": self.atomic_number,
+        # }
+
+    def dict(self, **kwargs) -> dict:
+        return self.json()
+
+    @classmethod
+    def parse_value_from_str_or_dict(cls, val):
```
Comment (deepanshs) [inline src/mrsimulator/spin_system/isotope.py:72]: Suggestions to try: rename this to `get_isotope` function which checks the `val` for str, dict, or Isotope types and return the Isotope object.

File: src/mrsimulator/method/__init__.py [hunk 2]
```diff
@@ -175,7 +175,14 @@ class Config:
...
     def validate_channels(cls, v, *, values, **kwargs):
-        return [Isotope(symbol=_) for _ in v]
+        lst = []
+        for val in v:
+            if isinstance(val, Isotope):
+                lst.append(val)
+
+            lst.append(Isotope.parse_value_from_str_or_dict(val))
+
+        return lst
```
Comment (deepanshs) [inline src/mrsimulator/method/__init__.py:185]: One line function:
```return [get_isotope(_) for _ in v]```

File: src/mrsimulator/method/lib/mqvas.py [hunk 3]
```diff
@@ -49,7 +49,11 @@ def update(cls, **kwargs):
 
         mq = MQ_p_symmetry[name]["mq"]
         iso = kwargs["channels"][0]
-        spin = Isotope(symbol=iso).spin if isinstance(iso, str) else iso.spin
+        spin = (
+            iso.spin
+            if isinstance(iso, Isotope)
+            else Isotope.parse_value_from_str_or_dict(iso).spin
+        )
```
Comment (deepanshs) [inline src/mrsimulator/method/lib/mqvas.py:56]: Maybe ```spin = get_isotope(iso).spin```

File: src/mrsimulator/spin_system/isotope.py [hunk 4]
```diff
@@ -14,6 +14,8 @@
 MODULE_DIR = path.dirname(path.abspath(__file__))
 ISOTOPE_DATA = loadfn(path.join(MODULE_DIR, "isotope_data.json"))
 
+custom_isotope_data = {}
```
Comment (deepanshs) [inline src/mrsimulator/spin_system/isotope.py:17]: Avoid using global variable. Make `custom_isotope_data` as a class attribute. All isotope objects will share the class attribute.

**Comment by mgiammar** [comment]
Current failing tests are checking `Method` object serialization. There is not a consensus yet on how to serialize the `channels` attribute with custom isotopes, and the PR will be updated once a decision has been reached.

**Comment by mgiammar** [comment]
After discussion, serializing each Isotope object in the `Method.channels` attribute as a dictionary makes the most sense since individual methods with custom isotope values can be serialized independent of `Site` or `SpinSystem` objects

**Comment by mgiammar** [comment]
Moved to #256

**Pull request closed without merging**